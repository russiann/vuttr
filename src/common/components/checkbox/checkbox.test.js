import React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Checkbox from './checkbox.component';

afterEach(cleanup);

it('renders', () => {
  const {asFragment} = render(<Checkbox />);
  expect(asFragment()).toMatchSnapshot();
});

it('should not show label if it dont pass label prop', () => {
  const {queryByTestId} = render(<Checkbox />);
  const label = queryByTestId('label');
  expect(label).toBeNull();
});

it('should show label', () => {
  const labelText = 'lorem ipsum';
  const {getByTestId} = render(<Checkbox label={labelText} />);
  const label = getByTestId('label');
  expect(label).toHaveTextContent(labelText);
});

it('should be checked if do not pass checked prop', () => {
  const {getByTestId} = render(<Checkbox onChange={jest.fn()} />);
  const checkbox = getByTestId('checkbox');
  expect(checkbox.checked).toBe(false);
});

it('should be checked if pass checked as true', () => {
  const {getByTestId} = render(<Checkbox checked onChange={jest.fn()} />);
  const checkbox = getByTestId('checkbox');
  expect(checkbox.checked).toBe(true);
});

it('should be checked if pass checked as false', () => {
  const {getByTestId} = render(
    <Checkbox checked={false} onChange={jest.fn()} />
  );
  const checkbox = getByTestId('checkbox');
  expect(checkbox.checked).toBe(false);
});
