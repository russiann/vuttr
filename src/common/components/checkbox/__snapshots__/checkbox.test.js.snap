// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <label
    class="css-1mj3kb8-CheckboxContainer e1ycchz00"
    data-testid="container"
  >
    <input
      class="css-ixs4ym-HiddenCheckbox e1ycchz02"
      data-testid="checkbox"
      type="checkbox"
    />
    <div
      class="css-3mta6a-StyledCheckbox e1ycchz03"
    >
      <svg
        class="css-dbk7ea-Icon e1ycchz01"
        viewBox="0 0 24 24"
      >
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </div>
  </label>
</DocumentFragment>
`;
